tools/snp.sh

```diff
@@ -459,14 +564,34 @@ build_guest_initrd() {
 }
 
 save_binary_paths() {
-  local guest_kernel=$(ls $(realpath "${SETUP_WORKING_DIR}/AMDSEV/linux/guest/debian/linux-image/boot/vmlinuz*"))
-  local guest_kernel_version=$(ls "${guest_kernel}" | sed "s|.*/boot/vmlinuz-\(.*\)|\1|g")
-  GENERATED_INITRD_BIN="${SETUP_WORKING_DIR}/initrd.img-${guest_kernel_version}"
+  [ -e /etc/os-release ] && . /etc/os-release
+
+  if [ "$ID" = "debian" ] || [ "$ID_LIKE" = "debian" ]; then
+    local guest_kernel=$(ls $(realpath "${SETUP_WORKING_DIR}/AMDSEV/linux/guest/debian/linux-image/boot/vmlinuz*"))
+    local guest_kernel_version=$(ls "${guest_kernel}" | sed "s|.*/boot/vmlinuz-\(.*\)|\1|g")
+    GENERATED_INITRD_BIN="${SETUP_WORKING_DIR}/initrd.img-${guest_kernel_version}"
+    OVMF_BIN="${SETUP_WORKING_DIR}/AMDSEV/ovmf/Build/AmdSev/DEBUG_GCC5/FV/OVMF.fd"
+
+  else
+    # Get snp guest version from kernel rpm pakage name
+    local guest_kernel_rpm_pckg=$(realpath ${SETUP_WORKING_DIR}/AMDSEV/linux/kernel-[0-9]*snp_guest*)
+    local rhel_kernel_name=$(basename "$guest_kernel_rpm_pckg")
+    local guest_kernel_version=$(echo "$rhel_kernel_name"| cut -d'-' -f2)
+
+    # copy and rename guest snp kernel from bzImage to vmlinuz
+    local uname_m=$(uname --m)
+    local bzImage_file=$(realpath ${SETUP_WORKING_DIR}/AMDSEV/linux/guest/arch/$uname_m/boot/bzImage)
```

Comment:
No. You cannot do this. `uname` will give you the kernel installed on the host. You need to detect this from some file or version in the guest folder.